spec/lib/submission_builder/ty2024/states/nj/documents/nj1040_spec.rb

```diff
@@ -229,5 +229,55 @@
         end
       end
     end
+
+    describe "property tax - lines 40a and 40b" do
+      context "when taxpayer is a renter" do
```

Comment:
[DUST] More of a testing best practices question - noticed that we're only testing MFS logic in the calculator, and nowhere else. To me this makes sense as the XML submission builder treats the values the same once they're saved to `calculated_fields`.

Is the takeaway here to not functionally test behavior in all scenarios for Calculator, XML, and PDF, but only in the classes that have relevant logic? (e.g. PDF filler and XML need to check a different box for homeowner vs renter, but not for MFS, it's just the calculated value that changes)

Reply:
When I'm doing TDD, it allows me to write exactly the right number of tests for the behavior under test. 

Under the TDD paradigm, there's no reason for me to write a test in the PDF or XML for the MFS scenario because it would not drive out in the PDF or XML classes themselves. I can't write a failing test to describe that. 

So yes, my roundabout answer is - there's no need to test unrelated functionality if I have a unit test of the XML or PDF behavior. TDD is nice because I don't have to actually _make that decision_ - I just follow the principle of writing a failing test first and if I can't write a failing test, then you could argue that behavior should not be under test here.